### PR TITLE
#1: Added to_dict

### PR DESCRIPTION
**Pull request by shacharoo**
Added the option to convert the struct to a dictionary.

### CODE REVIEW

File: src/basicstruct.py [hunk 1]
```diff
@@ -24,6 +25,20 @@ def __init__(self, *args, **kwargs):
             if not hasattr(self, key):
                 setattr(self, key, None)
 
+    def to_dict(self, copy=False):
+        """Convert the struct to a dictionary.
+
+        If `copy == True`, returns a `copy.deepcopy` of the values.
```
Comment (Nurdok) [inline src/basicstruct.py]: Just say "returns a deep copy of the values". The specific method is an implementation detail.

File: src/basicstruct.py [hunk 2]
```diff
@@ -62,6 +77,15 @@ def __ne__(self, other):
     def __hash__(self):
         return hash(self._to_tuple())
 
+    def __iter__(self):
+        """Yield pairs of (attrubute_name, value).
+
+        This allows using `dict(my_struct)`.
+
+        """
+        for key in self.__slots__:
```
Comment (Nurdok) [inline src/basicstruct.py]: Just `return iter(self._to_tuple())`

Comment (shacharoo) [inline src/basicstruct.py]: `dict(iterable)` expects pairs of `(key, value)`. `iter(self._to_tuple)` will yield only values.

Comment (Nurdok) [inline src/basicstruct.py]: Right, then `zip(self.__slots__, self._to_tuple())`

File: src/basicstruct.py [hunk 3]
```diff
@@ -62,6 +77,14 @@ def __ne__(self, other):
     def __hash__(self):
         return hash(self._to_tuple())
 
+    def __iter__(self):
+        """Yield pairs of (attrubute_name, value).
+
+        This allows using `dict(my_struct)`.
+
+        """
+        return izip(self.__slots__, self._to_tuple())
```
Comment (Nurdok) [inline src/basicstruct.py]: No, not `izip`. Notice that there's already an import of `zip` from `six`. It makes sure to use the correct, efficient `zip` for each Python version (essentially it import `izip` in Python 2.x and `zip` in Python 3.x).

**Pull request merged**